core/src/main/scala/com/nvidia/spark/rapids/tool/planparser/WholeStageExecParser.scala

```diff
@@ -45,20 +45,26 @@ case class WholeStageExecParser(
     val childNodes = node.nodes.flatMap { c =>
       SQLPlanParser.parsePlanNode(c, sqlID, checker, app, reusedNodeIds)
     }
-    // if any of the execs in WholeStageCodegen supported mark this entire thing
-    // as supported
+    // For the childNodes, we need to append the stages. Otherwise, nodes without metrics won't be
+    // assigned to stage
+    childNodes.foreach(_.appendToStages(stagesInNode))
+    // if any of the execs in WholeStageCodegen supported mark this entire thing as supported
     val anySupported = childNodes.exists(_.isSupported == true)
-    val unSupportedExprsArray = childNodes.filter(_.unsupportedExprs.length > 0 ).map(
-      x => x.unsupportedExprs).flatten.toArray
+    val unSupportedExprsArray =
+      childNodes.filter(_.unsupportedExprs.nonEmpty).flatMap(x => x.unsupportedExprs).toArray
     // average speedup across the execs in the WholeStageCodegen for now
     val supportedChildren = childNodes.filterNot(_.shouldRemove)
     val avSpeedupFactor = SQLPlanParser.averageSpeedup(supportedChildren.map(_.speedupFactor))
+    // TODO: revisit the logic behind adding the stages of child nodes to the current node.
```

Comment:
The below `allStagesIncludingChildren` is there since the early days. I don't quite understand the reason wholeStageCodeGen stageIDs are extended all the childNodes stages.
@tgravescs do you know why we this was needed? If not, I suggest we remove it.

Reply:
looking at the comment its saying that the child node might be associated to a stage but the whole stage codegen doesn't match it.  So we basically add all the child node ones in.    I don't remember when that can happen but would not remove it unless we are positive its not needed.

Reply:
so I don't think this comment was addressed before merged?  unless you know this doesn't happen I would prefer to see this comment removed or to file an issue to investigate it. This to me could just lead to confusion from someone reading the code.

Reply:
My bad! haven't got my cup of coffee. For some reason I misread that I got 2 approvals.

Reply:
Posted a followup https://github.com/NVIDIA/spark-rapids-tools/pull/1146